### ISSUE TITLE
Problem with training a SimpleRNN Network

### ISSUE THREAD

**Issue by Mahdizade** [fchollet/keras#1285]
Hi,
I using keras for near 2 month. I have no problem in training SimpleRNN networks in this period. Today I upgraded keras with sudo pip install --upgrade keras. After that I have a serious problem. I use a network with this config.

```
model = Sequential()
model.add(SimpleRNN(100, input_shape=X_train.shape[1:], activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(nb_classes))
model.add(Activation('softmax'))
sgd = SGD(lr=0.2, momentum=0.9, decay=0.0001)
model.compile(loss='categorical_crossentropy', optimizer=sgd)

model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=2, validation_data=(X_test, Y_test))
```
very simple network. but I get this output in terminal

```
Using Theano backend.
Using gpu device 0: GeForce GT 750M (CNMeM is enabled)
Train on 161488 samples, validate on 40372 samples
Epoch 1/40
12s - loss: 0.4373 - acc: 0.9183 - val_loss: 0.3246 - val_acc: 0.9442
Epoch 2/40
12s - loss: 0.4299 - acc: 0.9188 - val_loss: 0.3239 - val_acc: 0.9442
Epoch 3/40
13s - loss: 0.4296 - acc: 0.9188 - val_loss: 0.3235 - val_acc: 0.9442
Epoch 4/40
13s - loss: 0.4297 - acc: 0.9188 - val_loss: 0.3233 - val_acc: 0.9442
Epoch 5/40
13s - loss: 0.4296 - acc: 0.9188 - val_loss: 0.3254 - val_acc: 0.9442
Epoch 6/40
13s - loss: 0.4295 - acc: 0.9188 - val_loss: 0.3239 - val_acc: 0.9442
Epoch 7/40
13s - loss: 0.4296 - acc: 0.9188 - val_loss: 0.3243 - val_acc: 0.9442
Epoch 8/40
12s - loss: 0.4295 - acc: 0.9188 - val_loss: 0.3256 - val_acc: 0.9442
Epoch 9/40
12s - loss: 0.4295 - acc: 0.9188 - val_loss: 0.3260 - val_acc: 0.9442
Epoch 10/40
12s - loss: 0.4295 - acc: 0.9188 - val_loss: 0.3252 - val_acc: 0.9442
Epoch 11/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3258 - val_acc: 0.9442
Epoch 12/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3261 - val_acc: 0.9442
Epoch 13/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3228 - val_acc: 0.9442
Epoch 14/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3257 - val_acc: 0.9442
Epoch 15/40
13s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3242 - val_acc: 0.9442
Epoch 16/40
13s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3247 - val_acc: 0.9442
Epoch 17/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3259 - val_acc: 0.9442
Epoch 18/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3261 - val_acc: 0.9442
Epoch 19/40
12s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3254 - val_acc: 0.9442
Epoch 20/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3242 - val_acc: 0.9442
Epoch 21/40
13s - loss: 0.4294 - acc: 0.9188 - val_loss: 0.3251 - val_acc: 0.9442
Epoch 22/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3239 - val_acc: 0.9442
Epoch 23/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3245 - val_acc: 0.9442
Epoch 24/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3246 - val_acc: 0.9442
Epoch 25/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3249 - val_acc: 0.9442
Epoch 26/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3238 - val_acc: 0.9442
Epoch 27/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3278 - val_acc: 0.9442
Epoch 28/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3228 - val_acc: 0.9442
Epoch 29/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3254 - val_acc: 0.9442
Epoch 30/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3283 - val_acc: 0.9442
Epoch 31/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3247 - val_acc: 0.9442
Epoch 32/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3255 - val_acc: 0.9442
Epoch 33/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3264 - val_acc: 0.9442
Epoch 34/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3252 - val_acc: 0.9442
Epoch 35/40
14s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3267 - val_acc: 0.9442
Epoch 36/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3262 - val_acc: 0.9442
Epoch 37/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3249 - val_acc: 0.9442
Epoch 38/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3256 - val_acc: 0.9442
Epoch 39/40
12s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3244 - val_acc: 0.9442
Epoch 40/40
13s - loss: 0.4293 - acc: 0.9188 - val_loss: 0.3248 - val_acc: 0.9442
```
I can train the model before upgrading to new version. but after upgrading I have this problem in training any SimleRNN networks.  I see the same output after changing any parameters like optimizer, random seed and etc. 
I think something is wrong. I''ll so glad if someone say my mistake

**Comment by Mahdizade** [comment]
It solves with downgrade to version 0.2.0

**Comment by tboquet** [comment]
Did you tried with a smaller learning rate (`1e-4`) and/or a smaller decay (`1e-6`) for example?

**Comment by Mahdizade** [comment]
https://github.com/fchollet/keras/issues/1285#issuecomment-165155114
I try this now, no change.

**Comment by fchollet** [comment]
Try running on CPU to see if the problem is with the GPU.

**Comment by Mahdizade** [comment]
https://github.com/fchollet/keras/issues/1285#issuecomment-165162540
I try it before. on cpu or gpu, on theano or tensorflow backend, I get the same output

**Comment by fchollet** [comment]
Are you sure your version of Keras is up to date with the master branch?

**Comment by Mahdizade** [comment]
https://github.com/fchollet/keras/issues/1285#issuecomment-165180320
as I said, I upgrade from PyPI

**Comment by fchollet** [comment]
> as I said, I upgrade from PyPI

No, upgrade from the repo.

**Comment by Mahdizade** [comment]
https://github.com/fchollet/keras/issues/1285#issuecomment-165204795
no problem with repo version. thanks for your consideration
